lnrpc/routerrpc/router_server.go

```diff
@@ -649,7 +651,16 @@ func isLSP(routeHints [][]zpay32.HopHint) bool {
 		}
 	}
 
-	return true
+	// If the ref node hint contains a public channel we can send a probe to
```

Comment:
let's move it at the beginning so we can do the check first something like this:
We just check, does the refChanID exist, if it does we exit early.


```
_, _, _, err := routerBackend.FetchChannelInfo(refHint.ChannelID)
	if err == nil {
		return false
	}
```

Reply:
That makes sense, done.